### PR TITLE
feat: Feast AWS Athena offline store

### PR DESCRIPTION
**Pull request by toping4445**
What this PR does / why we need it:

It enables Feast users to use **AWS Athena** as an offline store.
I added a basic test module 
https://github.com/toping4445/feast/blob/master/sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/tests/test_athena_offline_store.py
refering to the code below
https://github.com/feast-dev/feast-custom-offline-store-demo/blob/main/test_custom_offline_store.py

However, due to the complexity of the configuration, I couldn't do the integration test.
The basic methods such as (get_historical_features) seem to work. Please let me know if you need to follow up after checking my codes.

Which issue(s) this PR fixes:

Fixes #

Does this PR introduce a user-facing change?:

I developed these modules to use Athena as an alternative to the Redshift(It is too expensive).
AWS Users can choose between Redshift and S3+Athena for an offline store.

### CODE REVIEW

File: protos/feast/core/DataSource.proto [hunk 1]
```diff
@@ -128,16 +128,14 @@ message DataSource {
...
+    string bootstrap_servers = 1;
 
     // Kafka topic to collect feature data from.
     string topic = 2;
 
     // Defines the stream data format encoding feature/entity data in Kafka messages.
     StreamFormat message_format = 3;
 
-    // Watermark delay threshold for stream data
-    google.protobuf.Duration watermark_delay_threshold = 4;
```
Comment (achals) [inline protos/feast/core/DataSource.proto:140]: Is there a reason this is being removed?

File: protos/feast/core/FeatureService.proto [hunk 2]
```diff
@@ -53,13 +54,15 @@ message FeatureServiceMeta {
 
 message LoggingConfig {
   float sample_rate = 1;
+  google.protobuf.Duration partition_interval = 2;
```
Comment (achals) [inline protos/feast/core/FeatureService.proto:57]: What does this represent?

File: sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/athena.py [hunk 3]
```diff
@@ -0,0 +1,692 @@
...
+import numpy as np
+import pandas as pd
+import pyarrow
+import pyarrow as pa
+from dateutil import parser
+from pydantic import StrictStr
+from pydantic.typing import Literal
+from pytz import utc
+
+#from feast import OnDemandFeatureView, AthenaSource
```
Comment (achals) [inline sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/athena.py:25]: rm?

File: sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/athena_source.py [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+
+class AthenaSource(DataSource):
+    def __init__(
+        self,
+        *,
+        timestamp_field: Optional[str] = "",
+        table: Optional[str] = None,
+        database: Optional[str] = None,
+        data_source: Optional[str] = None,
+        created_timestamp_column: Optional[str] = "",
```
Comment (achals) [inline sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/athena_source.py:38]: I would default this to `None`

File: protos/feast/core/DataSource.proto (hunk 1 above)
Comment (toping4445) [reply protos/feast/core/DataSource.proto:140]: Hello. sorry to come back late.
It seems to have been erased by overwriting my code which is based on previous version. 
I will fix it.

File: protos/feast/core/FeatureService.proto (hunk 2 above)
Comment (toping4445) [reply protos/feast/core/FeatureService.proto:57]: same issue. 
It seems those lines are deleted at new version.
I will fix them.

File: sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/athena.py (hunk 3 above)
Comment (toping4445) [reply sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/athena.py:25]: I will erase the unnecessary line.

File: sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/athena_source.py (hunk 4 above)
Comment (toping4445) [reply sdk/python/feast/infra/offline_stores/contrib/athena_offline_store/athena_source.py:38]: I will change it to None.

**Comment by toping4445** [comment]
I developed these add-ons based on the 0.21 version, so there were many conflicts. 
I erased all the parts that makes conflict. 
I'm sorry for my inexperience because it's my first open source commit.
If you give me feedback, I will make additional corrections

**Comment by toping4445** [comment]
1. I added AthenaDataSourceCreator for universal test.
2. I added a simple example with feature_repo.yaml file in the templates directory.
3. I modified the code format to meet the standard of feast.
4. There is actually no conflict in the file aws_utils.py 

make test-python
122 passed, 75 warnings, 11 errors in 61.05s
make test-python-universal
16 warnings, 4 errors in 8.67s

**Comment by kevjumba** [comment]
@toping4445 Hey Toping! Sorry about the silence on our side. We've been trying to get to the offline store contributions and were about to get to your athena offline store. We were actually adding documentation about adding offline stores to the repo to make this process a lot easier. 

If you would like to take a look here it is: https://github.com/feast-dev/feast/blob/master/docs/how-to-guides/adding-a-new-offline-store.md
If you would like to open this pr again, we will take a look soon!

**Pull request closed without merging**